.github/workflows/run-framework-tests.yaml

```diff
@@ -0,0 +1,53 @@
+on:
+  pull_request:
+    types: [opened, reopened]
+    branches:
+      - main
+
+name: run-framework-tests
+
+jobs:
+  run-framework-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Nest
+        uses: actions/checkout@v4
+        with:
+          path: ''
+          fetch-depth: 0 # Checkout everything to get access to the tags
+          repository: ${{ github.repository }}
+          ref: ${{github.event.pull_request.head.ref}}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout Flock
+        uses: actions/checkout@v4
+        with:
+          path: 'flock'
+          fetch-depth: 0 # Checkout everything to get access to the tags
+          repository: 'flutter/flutter'
+          ref: 'master'
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      #- name: Apply Nest patches to Flock
+      #  run: |
+      #    tools/git-import-patches patches
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Setup Android Tools
+        run: |
+          sudo apt-get install android-tools-adb
+
+      - name: Run Tests
+        run: |
+          PATH="$PATH:`pwd`/flock/bin"
+          flutter update-packages
```

Comment:
What does `update-packages` do?

Reply:
https://github.com/flutter/flutter/blob/master/docs/contributing/Setting-up-the-Framework-development-environment.md

flutter update-packages

This will recursively fetch all the Dart packages that Flutter depends on. If version solving failed, try git fetch upstream to update Flutter versions before flutter update-packages.